modules/ducktests/tests/ignitetest/tests/discovery_test.py

```diff
@@ -91,81 +96,57 @@ def setUp(self):
+    @matrix(ignite_version=[str(DEV_BRANCH), str(LATEST_2_8)],
+            kill_coordinator=[False, True],
+            nodes_to_kill=[0, 1, 2],
+            with_load=[False, True])
+    def test_tcp(self, ignite_version, kill_coordinator, nodes_to_kill, with_load):
         """
-        Test single node failure scenario (not the coordinator) with ZooKeeper.
+        Test nodes failure scenario with TcpDiscoverySpi.
         """
-        self.__start_zk_quorum()
-
-        return self.__simulate_nodes_failure(version, self.__zk_properties(self.zk_quorum.connection_string()), 1)
+        return self.__simulate_nodes_failure(ignite_version, self.__properties(), None, kill_coordinator,
+                                             nodes_to_kill, with_load)
 
     @cluster(num_nodes=NUM_NODES + 3)
-    @parametrize(version=str(DEV_BRANCH))
-    @parametrize(version=str(LATEST_2_7))
-    def test_zk_not_coordinator_two(self, version):
+    @matrix(ignite_version=[str(DEV_BRANCH), str(LATEST_2_8)],
+            kill_coordinator=[False, True],
+            nodes_to_kill=[0, 1, 2],
+            with_load=[False, True])
+    def test_zk(self, ignite_version, kill_coordinator, nodes_to_kill, with_load):
         """
-        Test two-node-failure scenario (not the coordinator) with ZooKeeper.
+        Test node failure scenario with ZooKeeperSpi.
         """
         self.__start_zk_quorum()
 
-        return self.__simulate_nodes_failure(version, self.__zk_properties(self.zk_quorum.connection_string()), 2)
+        properties = self.__zk_properties(self.zk_quorum.connection_string())
+        modules = ["zookeeper"]
 
-    @cluster(num_nodes=NUM_NODES+3)
-    @parametrize(version=str(DEV_BRANCH))
-    @parametrize(version=str(LATEST_2_7))
-    def test_zk_coordinator(self, version):
-        """
-        Test coordinator-failure scenario with ZooKeeper.
-        """
-        self.__start_zk_quorum()
+        return self.__simulate_nodes_failure(ignite_version, properties, modules, kill_coordinator, nodes_to_kill,
+                                             with_load)
 
-        return self.__simulate_nodes_failure(version, self.__zk_properties(self.zk_quorum.connection_string()), 0)
+    def __simulate_nodes_failure(self, version, properties, modules, kill_coordinator=False, nodes_to_kill=1,
+                                 with_load=False):
+        if nodes_to_kill == 0 and not kill_coordinator:
+            return {"No nodes to kill": "Nothing to do"}
 
-    def __simulate_nodes_failure(self, version, properties, nodes_to_kill=1):
         """
         :param nodes_to_kill: How many nodes to kill. If <1, the coordinator is the choice. Otherwise: not-coordinator
         nodes of given number.
         """
         self.servers = IgniteService(
             self.test_context,
-            num_nodes=self.NUM_NODES,
-            modules=["zookeeper"],
+            num_nodes=self.NUM_NODES - 1 if with_load else self.NUM_NODES,
```

Comment:
this will produce incomparable results. 
let's have the same clusters.

Reply:
Fixed